### PR TITLE
Use Generic type to make OpenFileEntry extensible

### PR DESCRIPTION
**Pull request by maobaolong**

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
  * Utility methods for Alluxio-FUSE.
  */
 @ThreadSafe
-public final class AlluxioFuseUtils {
+public class AlluxioFuseUtils {
```
Comment (jiacheliu3) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:42]: Do you want to extend `AlluxioFuseUtils`?

Comment (maobaolong) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:42]: Yeah, I have a project dependent alluxio-fuse, my Class want to extends `AlluxioFuseUtils `

**Comment by maobaolong** [comment]
@jiacheliu3 Thanks for your previous review, please take a look again

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java (hunk 1 above)
Comment (jiacheliu3) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:42]: It's not great to extend an utility class, so they are often made `final`. and only have static methods. I think you are not inheriting anything when you extend `AlluxioFuseUtils` right? I would actually recommend just having a separate class to keep the extra util methods.

Everything else looks good to me :)

Comment (maobaolong) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java]: @jiacheliu3 Thank you for your reply, It make sense to keep final

**Review by jiacheliu3 — APPROVED** [review]
@maobaolong Thanks for resolving the comments, LGTM

**Comment by maobaolong** [comment]
@jiacheliu3 I revert to keep final for AlluxioFuseUtils by the last commit. PTAL.

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**